core/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -140,6 +140,16 @@
         new Setting<>("node.mode", "network", Function.identity(), Property.NodeScope);
     public static final Setting<Boolean> NODE_INGEST_SETTING =
         Setting.boolSetting("node.ingest", true, Property.NodeScope);
+
+    /**
+     * controls whether the node is allowed to persist things like metadata to disk
+     * Note that this does not control whether the node stores actual indices (see
+     * {@link #NODE_DATA_SETTING}). However, if this is false, {@link #NODE_DATA_SETTING}
+     * and {@link #NODE_MASTER_SETTING} must also be false.
+     *
+     */
+    public static final Setting<Boolean> NODE_LOCAL_STORAGE_SETTING = Setting.boolSetting("node.local_storage", true, Property.NodeScope);
```

Comment:
I do not think we should have this setting. I think that for the case of tribe nodes, the sub-nodes can just use the node ID of the "parent" mixed (in a repeatable way, e.g., as a seed to the random base-64 UUID) with a combination of the tribe node name (the `t1` part of `tribe.t1.` in the settings) and the downstream cluster name.